AzureMonitorAgent/agent.py

```diff
@@ -279,19 +279,25 @@ def compare_and_copy_bin(src, dest):
         os.chmod(dest, stat.S_IXGRP | stat.S_IRGRP | stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IXOTH | stat.S_IROTH)
 
 def copy_amacoreagent_binaries():
-    amacoreagent_bin_local_path = os.getcwd() + "/amaCoreAgentBin/amacoreagent"
+    current_arch = platform.machine()
+    amacoreagent_bin_local_path = os.getcwd() + "/amaCoreAgentBin/amacoreagent_" + current_arch
     amacoreagent_bin = "/opt/microsoft/azuremonitoragent/bin/amacoreagent"
     compare_and_copy_bin(amacoreagent_bin_local_path, amacoreagent_bin)
 
-    liblz4x64_bin_local_path = os.getcwd() + "/amaCoreAgentBin/liblz4x64.so"
-    liblz4x64_bin = "/opt/microsoft/azuremonitoragent/bin/liblz4x64.so"
-    compare_and_copy_bin(liblz4x64_bin_local_path, liblz4x64_bin)
+    if current_arch == 'x86_64':
+        libgrpc_bin_local_path = os.getcwd() + "/amaCoreAgentBin/libgrpc_csharp_ext.x86_64.so"
+        libgrpc_bin = "/opt/microsoft/azuremonitoragent/bin/libgrpc_csharp_ext.x86_64.so"
+        compare_and_copy_bin(libgrpc_bin_local_path, libgrpc_bin)
 
-    libgrpc_bin_local_path = os.getcwd() + "/amaCoreAgentBin/libgrpc_csharp_ext.x64.so"
-    libgrpc_bin = "/opt/microsoft/azuremonitoragent/bin/libgrpc_csharp_ext.x64.so"
-    compare_and_copy_bin(libgrpc_bin_local_path, libgrpc_bin)
+        liblz4x64_bin_local_path = os.getcwd() + "/amaCoreAgentBin/liblz4x64.so"
```

Comment:
No lz4 for arm64?

Reply:
nope. Confirmed with Kennedy that they drop down to gzip for ARM64 right now.